src/deck/guides/3.0-upgrade.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Upgrade to Kong Gateway 3.x
+---
+
+Starting with {{site.base_gateway}} 3.0, paths using regular expressions must have a `~` prefix.
+The decK format version must be `3.0` to work with the new paths.
+
+If you [migrate your {{site.base_gateway}} database](/gateway/latest/upgrade/) from 2.8.x to 3.0, the paths in the database will have the `~` prefix added.
+This can cause configuration drift between state files and the database.
+
+Before using your state files to update the database, convert them into 3.0 format using `deck convert`.
+
+{:.important}
+> **Important**: Do not use `deck sync` with {{site.base_gateway}} 3.x before converting paths into 3.0 format.
+This will break all regex routing in 3.x.
+
+## decK command behavior with 3.x
+
+When running decK commands against {{site.base_gateway}} 3.x, take note of the following behavior:
+
+`deck dump`
+: Explicitly sets the format version to 3.0.
+It assumes that the paths have been correctly transformed, either via Kong migrations, manually, or through a `deck convert`.
+
+`deck diff`, `deck validate`, or `deck sync`
```

Comment:
Right now, the branch I'm working on is implementing these checks only for `sync` and `diff`. Should we add them for `validate` too? @hbagdi

Reply:
I pulled that info from the design brief. I can remove if that's not the intended state.

Reply:
You're right, that was a miss from my side, apologise! 

I just added it [here](https://github.com/Kong/deck/pull/753/commits/bcb1f7e801614e71619d2f38af4b95645f593f24), please note that this makes sense only when `validate` is run with the `--online` flag, otherwise there is no way to check for the Kong version running.